v-next/hardhat-node-test-runner/src/task-action.ts

```diff
@@ -67,7 +67,19 @@ const testWithHardhat: NewTaskActionFunction<TestActionArguments> = async (
   }
 
   const tsx = new URL(import.meta.resolve("tsx/esm"));
-  process.env.NODE_OPTIONS = `--import "${tsx.href}"`;
+  const imports = [tsx.href];
+
+  if (hre.globalOptions.coverage === true) {
+    // NOTE: Saving the environment variable to ensure node:test workers operate
+    // with the information that the global --coverage option is enabled
+    process.env.HARDHAT_COVERAGE = "true";
```

Comment:
This feels quite hacky to be fair. Maybe we should come up with a more general way of passing down the process flags to subprocesses.